cirq-superstaq/cirq_superstaq/job.py

```diff
@@ -378,24 +410,26 @@ def counts(
             qubit_indices: If provided, only include measurements counts of these qubits.
 
         Returns:
-            A dictionary containing the frequency counts of the measurements.
+            A dictionary containing the frequency counts of the measurements the job indexed by
+            `index` or a list of such dictionaries for each respective sub-job.
 
         Raises:
-            SuperstaqUnsuccessfulJobException: If the job failed or has been canceled or deleted.
-            SuperstaqServerException: If unable to get the results from the API.
+            gss.SuperstaqUnsuccessfulJobException: If the job failed or has been canceled or
+                deleted.
+            gss.SuperstaqServerException: If unable to get the results from the API.
             TimeoutError: If no results are available in the provided timeout interval.
         """
         time_waited_seconds: float = 0.0
-        while self.status() not in self.TERMINAL_STATES:
+        while self.status(index) not in self.TERMINAL_STATES:
```

Comment:
predates this pr, but we should probably change this to
```suggestion
        while (status := self.status(index)) not in self.TERMINAL_STATES:
```
and use `status` in the error message below, instead of making another api call (which could in really rare circumstances give use something like "Final status was DONE" if a job finishes between the two calls)